src/psyclone/tests/profile_test.py

```diff
@@ -0,0 +1,547 @@
+    # the ProfileStart calls using a regular expressions (\w*
+    # being the group name enclosed in "") group. Python will store
+    # those two groups and they can be accessed using the resulting
+    # re object.group(n).
+    correct_re = ("subroutine invoke.*"
+                  "use profile_mod, only: ProfileData.*"
+                  r"TYPE\(ProfileData\), save :: profile.*"
+                  r"call ProfileStart\(.*, \"(\w*)\",.*\).*"
+                  "do j.*"
+                  "do i.*"
+                  "call compute_cu_code.*"
+                  "end.*"
+                  "end.*"
+                  "call ProfileEnd.*"
+                  r"call ProfileStart\(.*, \"(\w*)\",.*\).*"
+                  "do j.*"
+                  "do i.*"
+                  "call compute_cu_code.*"
+                  "end.*"
+                  "end.*"
+                  "call ProfileEnd")
+
+    groups = re.search(correct_re, code, re.I)
+    assert groups is not None
+
+    # Check that the region names are indeed different: group(1)
+    # is the first kernel region name crated by PSyclone, and
+    # group(2) the name used in the second ProfileStart.
+    # Those names must be different (otherwise the profiling tool
+    # would likely combine the two different regions into one).
+    assert groups.group(1) != groups.group(2)
+
+
+# -----------------------------------------------------------------------------
+def test_profile_kernels_gocean1p0():
+    '''Check that all kernels are instrumented correctly
+    '''
+    Profiler.set_options([Profiler.KERNELS])
+    _, invoke = get_invoke("gocean1.0", "single_invoke_"
+                           "two_kernels.f90", 0)
+
+    # Convert the invoke to code, and remove all new lines, to make
+    # regex matching easier
+    code = str(invoke.gen()).replace("\n", "")
+
+    # Test that kernel profiling works in case of two kernel calls
+    # in a single invoke subroutine - i.e. we need to have one profile
+    # start call before two nested loops, and one profile end call
+    # after that:
+    correct_re = ("subroutine invoke.*"
+                  "use profile_mod, only: ProfileData.*"
+                  r"TYPE\(ProfileData\), save :: profile.*"
+                  r"TYPE\(ProfileData\), save :: profile.*"
+                  r"call ProfileStart\(.*, (?P<profile1>\w*)\).*"
+                  "do j.*"
+                  "do i.*"
+                  "call.*"
+                  "end.*"
+                  "end.*"
+                  r"call ProfileEnd\((?P=profile1)\).*"
+                  r"call ProfileStart\(.*, (?P<profile2>\w*)\).*"
+                  "do j.*"
+                  "do i.*"
+                  "call.*"
+                  "end.*"
+                  "end.*"
+                  r"call ProfileEnd\((?P=profile2)\)")
+    groups = re.search(correct_re, code, re.I)
+    assert groups is not None
+    # Check that the variables are different
+    assert groups.group(1) != groups.group(2)
```

Comment:
Again, please also check that the actual arguments passed to the profiling API are correct.